hrjobcontract/js/src/job-contract/controllers/form/form-general.js

```diff
@@ -52,8 +52,14 @@ define([
             });
 
             $scope.$watch('entity.details.period_end_date', function(){
-                $scope.dpDateStartMax = entityDetails.period_end_date ? moment(entityDetails.period_end_date).subtract(1, 'day').format() : '';
+                if (entityDetails.period_end_date) {
+                    $scope.dpDateStartMax = moment(entityDetails.period_end_date).subtract(1, 'day').format();  
+                } else {
+                    $scope.dpDateStartMax = '';
+                    entityDetails.end_reason = '';
+                }
                 $scope.duration = duration(entityDetails.period_start_date, entityDetails.period_end_date);
+
```

Comment:
Unnecessary new line. I would actually put it between the if and this assignment
```js
}

$scope.duration = // ...